PGM/src/main/java/tc/oc/pgm/autojoin/AutoJoinMatchModule.java

```diff
@@ -0,0 +1,101 @@
+package tc.oc.pgm.autojoin;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import tc.oc.commons.bukkit.settings.SettingManagerProvider;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.events.PlayerChangePartyEvent;
+import tc.oc.pgm.join.JoinMatchModule;
+import tc.oc.pgm.join.JoinMethod;
+import tc.oc.pgm.match.MatchModule;
+import tc.oc.pgm.match.MatchPlayer;
+import tc.oc.pgm.match.MatchScope;
+
+
+import javax.inject.Inject;
+
+/**
+ * New join feature that allows players to join without interfacing with GUI
+ * with an AutoJoinSetting that allows players to use the legacy join feature
+ * instead.
+ */
+@ListenerScope(MatchScope.LOADED)
+public class AutoJoinMatchModule extends MatchModule implements Listener {
+    private Set<MatchPlayer> joiningPlayers;
+    private final SettingManagerProvider settingManagerProvider;
+    private final JoinMatchModule joinMatchModule;
+
+    @Inject public AutoJoinMatchModule(SettingManagerProvider settingManagerProvider, JoinMatchModule joinMatchModule) {
+        this.joiningPlayers = new HashSet<>();
+        this.settingManagerProvider = settingManagerProvider;
+        this.joinMatchModule = joinMatchModule;
+    }
+
+    @Override
+    public void disable() {
+        joiningPlayers.clear();
+    }
+
+    // Checks if the player is eligible when they join
+    @EventHandler(priority = EventPriority.MONITOR)
+    public void playerJoin(final PlayerChangePartyEvent event) {
+        MatchPlayer player = event.getPlayer();
+
+        // Ignore if the match has started
+        if(match.hasStarted()) return;
+
+        // Remove the player if the player is leaving
+        if(event.getNewParty() == null) {
+            joiningPlayers.remove(player);
+            return;
+        }
+
+        //Ignore if player is going to participate in a match
+        if(event.getNewParty().isParticipatingType()) return;
+
+        /* Ignore if player is already known; case:
+         * Player joined a participating team
+         * Player left a participating team
+         */
+        // Check exists to only handle cases where Match has not started
+        if(joiningPlayers.contains(player)) return;
+
+        // Ignore if player explicitly chooses the legacy join feature
```

Comment:
Generally, only use comments when something isn't obvious. You can remove most of these comments.